lib/services/flowdock.ts

```diff
@@ -215,17 +209,77 @@ export class FlowdockService extends MessageService implements ServiceEmitter, S
-                        thread: resData.thread_id,
-                    }
+                    message: resData.id,
+                    thread: resData.thread_id,
+                    url: `https://www.flowdock.com/app/${org}/${body.flow}/threads/${resData.thread_id}`,
                 },
                 source: this.serviceName,
             };
         });
     }
 
     /**
+     * Search for recent private messages with our account that match on username and regex.
+     * @param username - scope of the private messages to search
+     * @param filter - Narrow our search to just matches
+     * @returns {Bluebird<string[]>} - Promise that resolves to the message strings
+     */
+    private fetchPrivateMessages(username: string, filter: RegExp): Promise<string[]> {
+        // Fetch the id then 1-1 history associated with the username
+        return this.fetchUserId(username)
+        .then((userId) => {
+            return this.fetchFromSession(`/private/${userId}/messages`)
+            .then((fetchedMessages) => {
+                // Prune and clean the message history to text of interest
+                return(_.filter(fetchedMessages, (message: FlowdockMessage) => {
+                    return filter.test(message.content);
+                }).map((message: FlowdockMessage) => {
+                    return message.content;
+                }));
+            });
+        });
+    }
+
+    /**
+     * Fetch a user's id from their username
+     * @param username - username to search for
+     * @returns {Bluebird<string>} - id of the user
+     */
+    private fetchUserId(username: string): Promise<string> {
+        // Get all the users of the service
+        return this.fetchFromSession(`/organizations/${process.env.FLOWDOCK_ORGANIZATION_NAME}/users`)
+        .then((foundUsers) => {
+            // Generate an array of user objects with matching username
+            const matchingUsers = _.filter(foundUsers, (eachUser: any) => {
+                return eachUser.nick === username;
+            });
+            // Return id if we've exactly one user for a particular username
+            if (matchingUsers.length === 1) {
+                return(matchingUsers[0].id);
+            } else {
+                throw new Error('Wrong number of users found in flowdock');
+            }
+        });
+    }
+
+    /**
+     * Utility function to structure the flowdock session as a promise a little
+     * @param path - Endpoint to retrieve
+     * @returns {Bluebird<any>} - response from the session
+     */
+    private fetchFromSession(path: string): Promise<any> {
+        return new Promise<any>((resolve, reject) => {
+            // The flowdock service both emits and calls back the error.
+            // We're wrapping the emit in a promise reject and ignoring the call back
+            FlowdockService.session.on('error', reject);
+            FlowdockService.session.get(path, {}, (_error?: Error, result?: any) => {
```

Comment:
I *really* do not want to be messing with the Flowdock session, it is schwonky enough code as it is.
I particularly dislike these lines:
```
          if (error != null) {
            _this.emit('error', error);
            return typeof cb === "function" ? cb(error) : void 0;
          } else {
            return typeof cb === "function" ? cb(null, body, res) : void 0;
          }
```
Which emit an error, and then if we handle it, also call back with the error just in case we missed it the first time.